share/man/man9/cdefs.9

```diff
@@ -0,0 +1,351 @@
+.Fx
+sources only.
+They are not intended as a general portability layer.
+.Sh Supported Compilers
+.Bl -tag -offset 2n -width 0
+.It Compilers supported for building programs on Fx :
+.Bl -column -offset 0n indent-two
+.It Sy Compiler Ta Sy Versions
+.It gcc Ta 9, 10, 11, 12, 13, 14
+.It clang Ta 10, 11, 12, 13, 14, 15, 16, 17, 18
+.It TinyC (tcc) Ta 0.9
+.It pcc Ta 1.1
+.El
+.Pp
+Due to testing constraints, tcc and pcc may not always work.
+.It Compilers supported for building Fx itself:
+.Bl -column -offset 0n indent-two
+.It Sy Compiler Ta Sy Versions
+.It gcc Ta 12, 13
+.It clang Ta 16, 17, 19
+.El
+.Sh Macros and Magic for Programming Environment
+.Nm
+defines (or refrains from defining) a number of macros to increase portability
+of compiled programs.
+These are to allow more advanced language features to appear in header files.
+The header files assume a compiler that accepts C prototype function
+declarations.
+They also assume that the compiler accepts ANSI C89 keywords for all language
+dialects.
+.Ss General Macros
+General macros that facilitate multiple language environments and language
+dialects.
+.Bl -column "---------------"
+.It Sy Macro Ta Sy Description
+.It Dv __volatile Ta expands to volatile in C++ and C89 and newer environments,
+__volatile in pre-ANSI environments that support this extension or nothing
+otherwise.
+.It Dv __inline Ta expands to inline in C++ and C89 and newer environments,
+__inline in pre-ANSI environments that support this extension or nothing
+otherwise.
+.It Dv __restrict Ta expands to restrict in C99 and newer environments, or
+__restrict otherwise.
+.It Dv __CONCAT Ta used to paste two pre-processor tokens.
+.It Dv __STRING Ta used to convert the argument to a string.
+.It Dv __BEGIN_DECLS Ta Start a group of functions.
+.It Dv __END_DECLS Ta End a group of functions.
+In a C environment, these are defined as nothing.
+In a C++ environment, these declare the functions to have
+.Dq C
+linkage.
+.El
+.Ss Function, Structure and Variable Modifiers
+.Bl -column "---------------"
+.It Sy Macro Ta Sy Description
+.It Sy __weak_symbol Ta Declare the symbol to be a weak symbol
+.It Sy __dead2 Ta Function will not return
+.It Sy __pure2 Ta Function has no side effects
+.It Sy __unused Ta To Variable may be unused (usually arguments), so do not warn about it
+.It Sy __used Ta Function really is used, so emit it even if it appears unused.
+.It Sy __packed Ta \&Do not have space between structure elements for natural alignment.
+Used when communicating with external protocols.
+.It Sy __aligned(x) Ta Specify in bytes the minimum alignment for the specified field, structure or variable
+.It Sy __section(x) Ta Place function or variable in section Fa x
+.It Sy __writeonly Ta Hint that the variable is only assigned to, but do not warn about it.
+Useful for macros and other places the eventual use of the result is unknown.
+.It Sy __alloc_size(x) Ta
+.It Sy __alloc_size2(x,n) Ta
+.It Sy __alloc_align(x) Ta
```

Comment:
```suggestion
.It Sy __alloc_align(x) Ta Hint pointer alignment specified by parameter Va x
```
Is this reasonable? I'm reading llvm docs and this is my first exposure to these concepts.